### DISCUSSION TITLE
How to load a custom dataset (COCO format) for instance segmentation?

### DISCUSSION THREAD

**Discussion by jeanpat** [Lightning-AI/pytorch-lightning#11503] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello,
I have a [dataset available on kaggle](https://www.kaggle.com/jeanpat/ultrasmallcoco-overlapping-chromosomes) consisting in 126 images and their segmentation labels in COCO format.

How can I modify [the given example](https://lightning-flash.readthedocs.io/en/stable/reference/instance_segmentation.html):
```
# 1. Create the DataModule
data_dir = icedata.pets.load_data()

datamodule = InstanceSegmentationData.from_folders(
    train_folder=data_dir,
    val_split=0.1,
    parser=partial(icedata.pets.parser, mask=True),
)
```

 to load it?
Would it be possible then to introduce simply data augmentation (translations..., rotations...)?

Thank you

**Comment by rohitgr7** [comment]
hey @jeanpat!

the ques is related to flash. Would you mind asking it here instead: https://github.com/PyTorchLightning/lightning-flash/discussions?
